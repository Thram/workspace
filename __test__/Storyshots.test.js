import initStoryshots from '@storybook/addon-storyshots';

// Mock Fetch
global.fetch = () => new Promise(resolve => resolve({ test: 'Test' }));

initStoryshots();
