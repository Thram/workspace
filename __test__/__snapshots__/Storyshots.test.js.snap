// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Storyshots Addons • Info 1`] = `
<div>
  <div
    style={
      Object {
        "position": "relative",
        "zIndex": 0,
      }
    }
  >
    <button
      className="pure-button  "
      onClick={[Function]}
      type="button"
    >
      Click Me!
    </button>
  </div>
  <a
    onClick={[Function]}
    style={
      Object {
        "background": "#28c",
        "borderRadius": "0 0 0 5px",
        "color": "#fff",
        "cursor": "pointer",
        "display": "block",
        "fontFamily": "sans-serif",
        "fontSize": "12px",
        "padding": "5px 15px",
        "position": "fixed",
        "right": 0,
        "textDecoration": "none",
        "top": 0,
      }
    }
  >
    Show Info
  </a>
  <div
    style={
      Object {
        "background": "white",
        "bottom": 0,
        "display": "none",
        "left": 0,
        "overflow": "auto",
        "padding": "0 40px",
        "position": "fixed",
        "right": 0,
        "top": 0,
      }
    }
  >
    <a
      onClick={[Function]}
      style={
        Object {
          "background": "#28c",
          "borderRadius": "0 0 0 5px",
          "color": "#fff",
          "cursor": "pointer",
          "display": "block",
          "fontFamily": "sans-serif",
          "fontSize": "12px",
          "padding": "5px 15px",
          "position": "fixed",
          "right": 0,
          "textDecoration": "none",
          "top": 0,
        }
      }
    >
      ×
    </a>
    <div
      style={undefined}
    >
      <div
        style={
          Object {
            "WebkitFontSmoothing": "antialiased",
            "color": "#444",
            "fontFamily": "
              -apple-system, \\".SFNSText-Regular\\", \\"San Francisco\\", \\"Roboto\\",
              \\"Segoe UI\\", \\"Helvetica Neue\\", \\"Lucida Grande\\", sans-serif
            ",
            "fontSize": "15px",
            "fontWeight": 300,
            "lineHeight": 1.45,
          }
        }
      >
        <div
          style={
            Object {
              "borderBottom": "1px solid #eee",
              "marginBottom": 10,
              "paddingTop": 10,
            }
          }
        >
          <h1
            style={
              Object {
                "fontSize": "35px",
                "margin": 0,
                "padding": 0,
              }
            }
          >
            Addons
          </h1>
          <h2
            style={
              Object {
                "fontSize": "22px",
                "fontWeight": 400,
                "margin": "0 0 10px 0",
                "padding": 0,
              }
            }
          >
            • Info
          </h2>
        </div>
        <div
          style={
            Object {
              "marginBottom": 0,
            }
          }
        >
          <p>
            This is the basic usage of the 
            <code>
              info addon
            </code>
            . Don’t forget to define your 
            <code>
              propTypes
            </code>
             .
          </p>
        </div>
        <div>
          <h1
            style={
              Object {
                "borderBottom": "1px solid #EEE",
                "fontSize": "25px",
                "margin": "20px 0 0 0",
                "padding": "0 0 5px 0",
              }
            }
          >
            Story Source
          </h1>
          <pre
            style={
              Object {
                "backgroundColor": "#fafafa",
                "fontFamily": "Menlo, Monaco, \\"Courier New\\", monospace",
                "fontSize": ".88em",
                "lineHeight": 1.5,
                "overflowX": "scroll",
                "padding": ".5rem",
              }
            }
          >
            <div>
              <div
                style={
                  Object {
                    "paddingLeft": 18,
                    "paddingRight": 3,
                  }
                }
              >
                <span
                  style={
                    Object {
                      "color": "#777",
                    }
                  }
                >
                  &lt;
                  Button
                </span>
                <span>
                  <span>
                     
                    <span
                      style={Object {}}
                    >
                      onClick
                    </span>
                    <span>
                      =
                      <span
                        style={Object {}}
                      >
                        <span>
                          <span
                            style={
                              Object {
                                "color": "#170",
                              }
                            }
                          >
                            Button_Clicked()
                          </span>
                        </span>
                      </span>
                    </span>
                    
                  </span>
                </span>
                <span
                  style={
                    Object {
                      "color": "#777",
                    }
                  }
                >
                  &gt;
                </span>
              </div>
              <div
                style={
                  Object {
                    "paddingLeft": 33,
                    "paddingRight": 3,
                  }
                }
              >
                <span
                  style={
                    Object {
                      "color": "#777",
                    }
                  }
                >
                  Click Me!
                </span>
              </div>
              <div
                style={
                  Object {
                    "paddingLeft": 18,
                    "paddingRight": 3,
                  }
                }
              >
                <span
                  style={
                    Object {
                      "color": "#777",
                    }
                  }
                >
                  &lt;/
                  Button
                  &gt;
                </span>
              </div>
            </div>
          </pre>
        </div>
        <div>
          <h1
            style={
              Object {
                "borderBottom": "1px solid #EEE",
                "fontSize": "25px",
                "margin": "20px 0 0 0",
                "padding": "0 0 5px 0",
              }
            }
          >
            Prop Types
          </h1>
          <div>
            <h2
              style={
                Object {
                  "margin": "20px 0 0 0",
                }
              }
            >
              "
              Button
              " Component
            </h2>
            <table
              style={
                Object {
                  "borderCollapse": "separate",
                  "borderSpacing": "10px 5px",
                  "marginLeft": -10,
                }
              }
            >
              <thead>
                <tr>
                  <th>
                    property
                  </th>
                  <th>
                    propType
                  </th>
                  <th>
                    required
                  </th>
                  <th>
                    default
                  </th>
                  <th>
                    description
                  </th>
                </tr>
              </thead>
              <tbody>
                <tr>
                  <td>
                    active
                  </td>
                  <td>
                    bool
                  </td>
                  <td>
                    no
                  </td>
                  <td>
                    <span>
                      <span
                        style={
                          Object {
                            "color": "#a11",
                          }
                        }
                      >
                        false
                      </span>
                    </span>
                  </td>
                  <td />
                </tr>
                <tr>
                  <td>
                    className
                  </td>
                  <td>
                    string
                  </td>
                  <td>
                    no
                  </td>
                  <td>
                    <span
                      style={
                        Object {
                          "color": "#22a",
                          "wordBreak": "break-word",
                        }
                      }
                    >
                      "
                      
                      "
                    </span>
                  </td>
                  <td />
                </tr>
                <tr>
                  <td>
                    disabled
                  </td>
                  <td>
                    bool
                  </td>
                  <td>
                    no
                  </td>
                  <td>
                    <span>
                      <span
                        style={
                          Object {
                            "color": "#a11",
                          }
                        }
                      >
                        false
                      </span>
                    </span>
                  </td>
                  <td />
                </tr>
                <tr>
                  <td>
                    hidden
                  </td>
                  <td>
                    bool
                  </td>
                  <td>
                    no
                  </td>
                  <td>
                    <span>
                      <span
                        style={
                          Object {
                            "color": "#a11",
                          }
                        }
                      >
                        false
                      </span>
                    </span>
                  </td>
                  <td />
                </tr>
                <tr>
                  <td>
                    hover
                  </td>
                  <td>
                    bool
                  </td>
                  <td>
                    no
                  </td>
                  <td>
                    <span>
                      <span
                        style={
                          Object {
                            "color": "#a11",
                          }
                        }
                      >
                        false
                      </span>
                    </span>
                  </td>
                  <td />
                </tr>
                <tr>
                  <td>
                    selected
                  </td>
                  <td>
                    bool
                  </td>
                  <td>
                    no
                  </td>
                  <td>
                    <span>
                      <span
                        style={
                          Object {
                            "color": "#a11",
                          }
                        }
                      >
                        false
                      </span>
                    </span>
                  </td>
                  <td />
                </tr>
                <tr>
                  <td>
                    size
                  </td>
                  <td>
                    string
                  </td>
                  <td>
                    no
                  </td>
                  <td>
                    -
                  </td>
                  <td />
                </tr>
                <tr>
                  <td>
                    submit
                  </td>
                  <td>
                    bool
                  </td>
                  <td>
                    no
                  </td>
                  <td>
                    <span>
                      <span
                        style={
                          Object {
                            "color": "#a11",
                          }
                        }
                      >
                        false
                      </span>
                    </span>
                  </td>
                  <td />
                </tr>
                <tr>
                  <td>
                    type
                  </td>
                  <td>
                    string
                  </td>
                  <td>
                    no
                  </td>
                  <td>
                    -
                  </td>
                  <td />
                </tr>
              </tbody>
            </table>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Storyshots Addons • Knobs 1`] = `
<div
  style={
    Object {
      "alignItems": "center",
      "background": "#5CACC4",
      "borderRadius": "50%",
      "display": "flex",
      "height": "100px",
      "margin": "1rem",
      "padding": "1rem",
      "textAlign": "center",
      "width": "100px",
    }
  }
>
  Edit my Styles!!
</div>
`;

exports[`Storyshots Components • Flat Button 1`] = `
<div
  className="pure-g css-d1ao1r "
  style={
    Object {
      "padding": "2rem",
    }
  }
>
  <div
    className="pure-u-1 css-d1ao1r "
  >
    <a
      onClick={[Function]}
      role="button"
      style={
        Object {
          "cursor": "pointer",
          "display": "inline-block",
          "width": "auto",
        }
      }
      tabIndex="0"
    >
      <div
        style={
          Object {
            "alignItems": "center",
            "display": "flex",
          }
        }
      >
        Text
      </div>
    </a>
    <a
      onClick={[Function]}
      role="button"
      style={
        Object {
          "cursor": "pointer",
          "display": "inline-block",
          "width": "auto",
        }
      }
      tabIndex="0"
    >
      <div
        style={
          Object {
            "alignItems": "center",
            "display": "flex",
            "width": "100px",
          }
        }
      >
        <img
          alt="Image"
          className="pure-img "
          src="https://s-media-cache-ak0.pinimg.com/736x/e5/a0/69/e5a06942fa42823c88be5f3a834e063d.jpg"
        />
      </div>
    </a>
  </div>
</div>
`;

exports[`Storyshots Components • Icon 1`] = `
<div
  className="pure-g css-d1ao1r "
  style={
    Object {
      "padding": "2rem",
    }
  }
>
  <div
    className="pure-u-1 css-d1ao1r "
  >
    <h1>
      Material Icons!
    </h1>
    <i
      className="material-icons"
      style={
        Object {
          "color": "#E6867A",
        }
      }
    >
      done
    </i>
    <i
      className="material-icons"
      style={
        Object {
          "color": "#ED4A6A",
          "fontSize": "100px",
        }
      }
    >
      extension
    </i>
    <i
      className="material-icons"
      style={
        Object {
          "color": "#D9AB91",
          "fontSize": "50px",
        }
      }
    >
      face
    </i>
    <i
      className="material-icons"
      style={
        Object {
          "color": "#D9D5A7",
          "fontSize": "30px",
        }
      }
    >
      motorcycle
    </i>
    <i
      className="material-icons"
      style={
        Object {
          "color": "#C7FCD7",
          "fontSize": "80px",
        }
      }
    >
      pets
    </i>
  </div>
</div>
`;

exports[`Storyshots Components • Link 1`] = `
<div
  className="pure-g css-d1ao1r "
  style={
    Object {
      "padding": "2rem",
    }
  }
>
  <div
    className="pure-u-1 css-d1ao1r "
  >
    <button
      className="pure-button pure-button-primary  "
      onClick={[Function]}
      type="button"
    >
      Click me!
    </button>
  </div>
</div>
`;

exports[`Storyshots Components • Todo List 1`] = `
<div
  className="pure-g css-d1ao1r "
  style={
    Object {
      "padding": "2rem",
    }
  }
>
  <div
    className="pure-u-1 css-d1ao1r "
  >
    <table
      className="pure-table pure-u-12-24 "
    >
      <tbody>
        <tr
          className=""
        >
          <td>
            <a
              onClick={[Function]}
              role="button"
              style={
                Object {
                  "cursor": "pointer",
                  "display": "inline-block",
                  "width": "auto",
                }
              }
              tabIndex="0"
            >
              <div
                style={
                  Object {
                    "alignItems": "center",
                    "display": "flex",
                  }
                }
              >
                <i
                  className="material-icons"
                >
                  check_box_outline_blank
                </i>
                <span
                  style={
                    Object {
                      "marginLeft": "1rem",
                    }
                  }
                >
                  Task 1
                </span>
              </div>
            </a>
          </td>
        </tr>
        <tr
          className=""
        >
          <td>
            <a
              onClick={[Function]}
              role="button"
              style={
                Object {
                  "cursor": "pointer",
                  "display": "inline-block",
                  "width": "auto",
                }
              }
              tabIndex="0"
            >
              <div
                style={
                  Object {
                    "alignItems": "center",
                    "display": "flex",
                  }
                }
              >
                <i
                  className="material-icons"
                >
                  check_box_outline_blank
                </i>
                <span
                  style={
                    Object {
                      "marginLeft": "1rem",
                    }
                  }
                >
                  Task 2
                </span>
              </div>
            </a>
          </td>
        </tr>
        <tr
          className="pure-table-odd"
        >
          <td>
            <a
              onClick={[Function]}
              role="button"
              style={
                Object {
                  "cursor": "pointer",
                  "display": "inline-block",
                  "width": "auto",
                }
              }
              tabIndex="0"
            >
              <div
                style={
                  Object {
                    "alignItems": "center",
                    "display": "flex",
                  }
                }
              >
                <i
                  className="material-icons"
                >
                  check_box
                </i>
                <span
                  style={
                    Object {
                      "marginLeft": "1rem",
                    }
                  }
                >
                  Task 3
                </span>
              </div>
            </a>
          </td>
        </tr>
        <tr
          className=""
        >
          <td>
            <a
              onClick={[Function]}
              role="button"
              style={
                Object {
                  "cursor": "pointer",
                  "display": "inline-block",
                  "width": "auto",
                }
              }
              tabIndex="0"
            >
              <div
                style={
                  Object {
                    "alignItems": "center",
                    "display": "flex",
                  }
                }
              >
                <i
                  className="material-icons"
                >
                  check_box_outline_blank
                </i>
                <span
                  style={
                    Object {
                      "marginLeft": "1rem",
                    }
                  }
                >
                  Task 4
                </span>
              </div>
            </a>
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;

exports[`Storyshots Containers • Add Todo 1`] = `
<div
  className="pure-g css-d1ao1r "
  style={
    Object {
      "padding": "2rem",
    }
  }
>
  <div
    className="pure-u-1 css-d1ao1r "
  >
    <div
      className="pure-g css-d1ao1r "
    >
      <div
        className="pure-u-1 css-d1ao1r "
      >
        <form
          className="pure-form pure-form-aligned "
          onSubmit={[Function]}
        >
          <div
            className="pure-control-group "
          >
            <input
              className="pure-input "
              style={
                Object {
                  "marginRight": "1rem",
                }
              }
            />
            <button
              className="pure-button  "
              type="submit"
            >
              Add Todo
            </button>
          </div>
        </form>
      </div>
    </div>
  </div>
</div>
`;

exports[`Storyshots Containers • Filter 1`] = `
<div
  className="pure-g css-d1ao1r "
  style={
    Object {
      "padding": "2rem",
    }
  }
>
  <div
    className="pure-u-1 css-d1ao1r "
  >
    <div
      className="pure-g css-d1ao1r "
    >
      <div
        className="pure-u-1 css-d1ao1r "
      >
        Show:
        <div
          className="pure-button-group "
        >
          <button
            className="pure-button pure-button-primary  "
            onClick={[Function]}
            type="button"
          >
            All
          </button>
          <button
            className="pure-button pure-button-secondary  "
            onClick={[Function]}
            type="button"
          >
            Active
          </button>
          <button
            className="pure-button pure-button-success  "
            onClick={[Function]}
            type="button"
          >
            Completed
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Storyshots Containers • Filter Link 1`] = `
<div
  className="pure-g css-d1ao1r "
  style={
    Object {
      "padding": "2rem",
    }
  }
>
  <div
    className="pure-u-1 css-d1ao1r "
  >
    <button
      className="pure-button pure-button-primary  "
      onClick={[Function]}
      type="button"
    >
      Show All!!
    </button>
  </div>
</div>
`;

exports[`Storyshots Containers • Navigation 1`] = `
<div
  className="pure-g css-d1ao1r "
  style={
    Object {
      "padding": "2rem",
    }
  }
>
  <div
    className="pure-u-1 css-d1ao1r "
  >
    <div
      className="pure-g css-d1ao1r "
    >
      <div
        className="pure-u-1 css-d1ao1r "
      >
        <div
          style={
            Object {
              "alignItems": "center",
              "display": "flex",
            }
          }
        >
          <h2>
            Test
          </h2>
          <div
            style={
              Object {
                "marginLeft": "1rem",
              }
            }
          >
            <a
              className="active"
              href="#/"
              onClick={[Function]}
              style={
                Object {
                  "padding": "0.2rem 0.5rem",
                }
              }
            >
              Home
            </a>
            <a
              className={undefined}
              href="#/mocks"
              onClick={[Function]}
              style={
                Object {
                  "padding": "0.2rem 0.5rem",
                }
              }
            >
              Mocks
            </a>
            <a
              className={undefined}
              href="#/icons"
              onClick={[Function]}
              style={
                Object {
                  "padding": "0.2rem 0.5rem",
                }
              }
            >
              Icons
            </a>
          </div>
        </div>
      </div>
    </div>
    <div
      className="pure-g css-d1ao1r "
    >
      <div
        className="pure-u-1 css-d1ao1r "
      >
        <div>
          Home
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Storyshots Containers • Visible Todo List 1`] = `
<div
  className="pure-g css-d1ao1r "
  style={
    Object {
      "padding": "2rem",
    }
  }
>
  <div
    className="pure-u-1 css-d1ao1r "
  >
    <table
      className="pure-table pure-u-12-24 "
    >
      <tbody>
        <tr
          className=""
        >
          <td>
            <a
              onClick={[Function]}
              role="button"
              style={
                Object {
                  "cursor": "pointer",
                  "display": "inline-block",
                  "width": "auto",
                }
              }
              tabIndex="0"
            >
              <div
                style={
                  Object {
                    "alignItems": "center",
                    "display": "flex",
                  }
                }
              >
                <i
                  className="material-icons"
                >
                  check_box_outline_blank
                </i>
                <span
                  style={
                    Object {
                      "marginLeft": "1rem",
                    }
                  }
                >
                  Task 1
                </span>
              </div>
            </a>
          </td>
        </tr>
        <tr
          className=""
        >
          <td>
            <a
              onClick={[Function]}
              role="button"
              style={
                Object {
                  "cursor": "pointer",
                  "display": "inline-block",
                  "width": "auto",
                }
              }
              tabIndex="0"
            >
              <div
                style={
                  Object {
                    "alignItems": "center",
                    "display": "flex",
                  }
                }
              >
                <i
                  className="material-icons"
                >
                  check_box_outline_blank
                </i>
                <span
                  style={
                    Object {
                      "marginLeft": "1rem",
                    }
                  }
                >
                  Task 2
                </span>
              </div>
            </a>
          </td>
        </tr>
        <tr
          className="pure-table-odd"
        >
          <td>
            <a
              onClick={[Function]}
              role="button"
              style={
                Object {
                  "cursor": "pointer",
                  "display": "inline-block",
                  "width": "auto",
                }
              }
              tabIndex="0"
            >
              <div
                style={
                  Object {
                    "alignItems": "center",
                    "display": "flex",
                  }
                }
              >
                <i
                  className="material-icons"
                >
                  check_box
                </i>
                <span
                  style={
                    Object {
                      "marginLeft": "1rem",
                    }
                  }
                >
                  Task 3
                </span>
              </div>
            </a>
          </td>
        </tr>
        <tr
          className=""
        >
          <td>
            <a
              onClick={[Function]}
              role="button"
              style={
                Object {
                  "cursor": "pointer",
                  "display": "inline-block",
                  "width": "auto",
                }
              }
              tabIndex="0"
            >
              <div
                style={
                  Object {
                    "alignItems": "center",
                    "display": "flex",
                  }
                }
              >
                <i
                  className="material-icons"
                >
                  check_box_outline_blank
                </i>
                <span
                  style={
                    Object {
                      "marginLeft": "1rem",
                    }
                  }
                >
                  Task 4
                </span>
              </div>
            </a>
          </td>
        </tr>
      </tbody>
    </table>
    <div
      className="pure-g css-d1ao1r "
    >
      <div
        className="pure-u-1 css-d1ao1r "
      >
        Show:
        <div
          className="pure-button-group "
        >
          <button
            className="pure-button pure-button-primary  "
            onClick={[Function]}
            type="button"
          >
            All
          </button>
          <button
            className="pure-button pure-button-secondary  "
            onClick={[Function]}
            type="button"
          >
            Active
          </button>
          <button
            className="pure-button pure-button-success  "
            onClick={[Function]}
            type="button"
          >
            Completed
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Storyshots Views • Mocks 1`] = `
<div
  className="pure-g css-d1ao1r "
  style={
    Object {
      "padding": "2rem",
    }
  }
>
  <div
    className="pure-u-1 css-d1ao1r "
  >
    <div
      className="pure-g css-d1ao1r "
    >
      <div
        className="pure-u-1 css-d1ao1r "
      >
        <div
          className="pure-g css-d1ao1r "
        >
          <div
            className="pure-u-1 css-d1ao1r "
          >
            <div
              className="pure-button-group "
            >
              <button
                className="pure-button  "
                onClick={[Function]}
                type="button"
              >
                Users
              </button>
              <button
                className="pure-button  "
                onClick={[Function]}
                type="button"
              >
                Posts
              </button>
              <button
                className="pure-button  "
                onClick={[Function]}
                type="button"
              >
                Albums
              </button>
            </div>
          </div>
        </div>
        <pre
          style={
            Object {
              "whiteSpace": "pre-wrap",
            }
          }
        >
          Loading...
        </pre>
      </div>
    </div>
  </div>
</div>
`;

exports[`Storyshots Views • Todos 1`] = `
<div
  className="pure-g css-d1ao1r "
  style={
    Object {
      "padding": "2rem",
    }
  }
>
  <div
    className="pure-u-1 css-d1ao1r "
  >
    <div
      className="pure-g css-d1ao1r "
    >
      <div
        className="pure-u-1 css-d1ao1r "
      >
        <div
          className="pure-g css-d1ao1r "
        >
          <div
            className="pure-u-1 css-d1ao1r "
          >
            <form
              className="pure-form pure-form-aligned "
              onSubmit={[Function]}
            >
              <div
                className="pure-control-group "
              >
                <input
                  className="pure-input "
                  style={
                    Object {
                      "marginRight": "1rem",
                    }
                  }
                />
                <button
                  className="pure-button  "
                  type="submit"
                >
                  Add Todo
                </button>
              </div>
            </form>
          </div>
        </div>
        <table
          className="pure-table pure-u-12-24 "
        >
          <tbody />
        </table>
        <div
          className="pure-g css-d1ao1r "
        >
          <div
            className="pure-u-1 css-d1ao1r "
          >
            Show:
            <div
              className="pure-button-group "
            >
              <button
                className="pure-button pure-button-primary  "
                onClick={[Function]}
                type="button"
              >
                All
              </button>
              <button
                className="pure-button pure-button-secondary  "
                onClick={[Function]}
                type="button"
              >
                Active
              </button>
              <button
                className="pure-button pure-button-success  "
                onClick={[Function]}
                type="button"
              >
                Completed
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
